pkg/bisect/linux.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package bisect
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/pkg/vcs"
+	"github.com/google/syzkaller/prog"
+)
+
+var configBisectTag string = "# Minimized by syzkaller"
+
+func (env *env) minimizeConfig() {
```

Comment:
You moved it to a `linux.go` file, but it's still not linux-specific, it will be invoked for all OSes. Also not possible to stub for testing and not testable in itself.

I think we need something along the following lines in vcs package (similar to the Bisecter interface):
```
type ConfigBisecter interface {
	Bisect(bad, good []byte, trace io.Writer, pred func(test []byte) (bool, error)) ([]byte, error)
}
```

Then we will be able to check if the current OS implements it as:
`configBisecter, ok := repo.(vcs.ConfigBisecter)`
and more importantly we will be able to easily stub it for testing as we do with the Bisecter interface.
Last but not least, it will be possible to easily wrap it in a standalone test tool that will test the config minimization itself.

Reply:
I think I now got what you mean. Please check revised code. Thank you for patience.